gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/TokenStateDatabase.java

```diff
@@ -192,7 +196,9 @@ Collection<KnoxToken> getDoAsTokens(String userName) throws SQLException {
   private Collection<KnoxToken> fetchTokens(String userName, String sql) throws SQLException {
     Map<String, KnoxToken> tokenMap = new LinkedHashMap<>();
     try (Connection connection = dataSource.getConnection(); PreparedStatement getTokenIdsStatement = connection.prepareStatement(sql)) {
-      getTokenIdsStatement.setString(1, userName);
+      if (userName != null) {
+        getTokenIdsStatement.setString(1, userName);
```

Comment:
What happens if we don't set the userName parameter of the prepared statement? Does it work too?

Reply:
An SQL error is thrown here which is handled in `JDBCTokenStateService`:
```2023-10-09 11:25:29,822 84376fc3-8a56-42f3-a399-9f81f0ffe2da ERROR token.state (JDBCTokenStateService.java:getTokens(312)) - An error occurred while fetching tokens for user null from the database : No value specified for parameter 1.```

Reply:
The result is an empty token list in the response for `getUserTokens`.